### PR TITLE
SOLR-13802: Write analyzer property luceneMatchVersion to managed schema

### PR DESCRIPTION
**Pull request by thomaswoeckinger**
# Description

Please provide a short description of the changes you're making with this pull request.

# Solution

Please provide a short description of the approach taken to implement your solution.

# Tests

Please describe the tests you've developed or run to confirm this patch implements the feature or solves the problem.

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I am authorized to contribute this code to the ASF and have removed any code I do not have a license to distribute.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `master` branch.
- [x] I have run `ant precommit` and the appropriate test suite.
- [x] I have added tests for my changes.
- [ ] I have added documentation for the [Ref Guide](https://github.com/apache/lucene-solr/tree/master/solr/solr-ref-guide) (for Solr changes only).

### CODE REVIEW

**Comment by thomaswoeckinger** [comment]
@dsmiley: Please review

File: solr/core/src/test/org/apache/solr/rest/schema/TestBulkSchemaAPI.java [hunk 1]
```diff
@@ -125,7 +126,7 @@ public void testAnalyzerClass() throws Exception {
         "    'name' : 'myNewTextFieldWithAnalyzerClass',\n" +
         "    'class':'solr.TextField',\n" +
         "    'analyzer' : {\n" +
-        "        'luceneMatchVersion':'5.0.0',\n" +
+        "        '" + IndexSchema.LUCENE_MATCH_VERSION_PARAM + "':'5.0.0',\n" +
```
Comment (dsmiley) [inline solr/core/src/test/org/apache/solr/rest/schema/TestBulkSchemaAPI.java]: Using Constants in tests like here is a bit too far IMO.  It slightly obscures readability and there's a point to be made that changing the input/output constant _should_ break a test.  Subjective, I know.   I don't mean to suggest the removal of all constants in tests but here in particular it's ugly.

File: solr/core/src/test/org/apache/solr/rest/schema/TestSerializedLuceneMatchVersion.java [hunk 2]
```diff
@@ -44,13 +45,13 @@ public void testExplicitLuceneMatchVersions() throws Exception {
             "count(/response/lst[@name='fieldType']) = 1",
         
             "//lst[str[@name='class'][.='org.apache.solr.analysis.MockCharFilterFactory']]"
-           +"     [str[@name='luceneMatchVersion'][.='4.0.0']]",
+           +"     [str[@name='" + IndexSchema.LUCENE_MATCH_VERSION_PARAM + "'][.='4.0.0']]",
```
Comment (dsmiley) [inline solr/core/src/test/org/apache/solr/rest/schema/TestSerializedLuceneMatchVersion.java]: again; here in particular it's ugly

File: solr/core/src/test/org/apache/solr/analysis/TestWordDelimiterFilterFactory.java [hunk 3]
```diff
@@ -222,7 +223,7 @@ public void testCustomTypes() throws Exception {
     /* custom behavior */
     args = new HashMap<>();
     // use a custom type mapping
-    args.put("luceneMatchVersion", Version.LATEST.toString());
+    args.put(IndexSchema.LUCENE_MATCH_VERSION_PARAM, Version.LATEST.toString());
```
Comment (dsmiley) [inline solr/core/src/test/org/apache/solr/analysis/TestWordDelimiterFilterFactory.java]: not ugly (see later comment)

Comment (thomaswoeckinger) [reply solr/core/src/test/org/apache/solr/analysis/TestWordDelimiterFilterFactory.java]: later comment?

**Comment by thomaswoeckinger** [comment]
Seems that ant precommit check is toggling, locally all is working fine, read about problems in dev list

File: solr/core/src/test/org/apache/solr/analysis/TestWordDelimiterFilterFactory.java [hunk 4]
```diff
@@ -198,7 +199,7 @@ public void testCustomTypes() throws Exception {
     String testText = "I borrowed $5,400.00 at 25% interest-rate";
     ResourceLoader loader = new SolrResourceLoader(TEST_PATH().resolve("collection1"));
     Map<String,String> args = new HashMap<>();
-    args.put("luceneMatchVersion", Version.LATEST.toString());
+    args.put(IndexSchema.LUCENE_MATCH_VERSION_PARAM, Version.LATEST.toString());
```
Comment (dsmiley) [inline solr/core/src/test/org/apache/solr/analysis/TestWordDelimiterFilterFactory.java]: I'm going to remove these diffs in this file for consistency reasons.  All string literals is very clear.

**Pull request closed without merging**